src/overlays/actors/ovl_Dm_Char06/z_dm_char06.c

```diff
@@ -32,18 +30,60 @@ const ActorInit Dm_Char06_InitVars = {
     (ActorFunc)DmChar06_Draw,
 };
 
-#endif
+void DmChar06_SetupAction(DmChar06* this, DmChar06ActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void DmChar06_Init(Actor* thisx, PlayState* play) {
+    DmChar06* this = THIS;
+
+    gSaveContext.save.weekEventReg[33] |= 0x80;
+    Actor_SetScale(&this->actor, 1.0f);
+    this->alpha = 255;
+    DmChar06_SetupAction(this, func_80AAE6F0);
+}
+
+void DmChar06_Destroy(Actor* thisx, PlayState* play) {
+}
+
+void func_80AAE6F0(DmChar06* this, PlayState* play) {
+    if (Cutscene_CheckActorAction(play, 463)) {
+        s32 actionIndex = Cutscene_GetActorActionIndex(play, 0x1CF);
+
+        if ((play->csCtx.frames >= play->csCtx.actorActions[actionIndex]->startFrame) &&
+            (play->csCtx.actorActions[actionIndex]->endFrame >= play->csCtx.frames)) {
+            if (play->csCtx.actorActions[actionIndex]->action == 1) {
+                this->alpha = 255;
+            } else if (play->csCtx.actorActions[actionIndex]->action == 2) {
+                f32 floatTemp = 1.0f - Environment_LerpWeight(play->csCtx.actorActions[actionIndex]->endFrame,
```

Comment:
```suggestion
                f32 lerp= 1.0f - Environment_LerpWeight(play->csCtx.actorActions[actionIndex]->endFrame,
```
I think lerp would be a better variable name